peer/peer.go

```diff
@@ -1098,6 +1102,24 @@ func (p *Peer) readMessage(encoding wire.MessageEncoding) (wire.Message, []byte,
 	return msg, buf, nil
 }
 
+// SubscribeRecvMsg adds a OnRead subscription to the peer.  All bitcoin
+// messages received from this peer will be sent on the returned
+// channel.  A closure is also returned, that should be called to cancel
+// the subscription.
+func (p *Peer) SubscribeRecvMsg() (<-chan wire.Message, func()) {
+	msgChan := make(chan wire.Message, 1)
```

Comment:
Hmm, so a buffer of 1 will mean that `writeMessage` will block anytime a subscriber fails to immediately read out a message. If that goroutine blocks, then all outbound message writing will also halt.

Reply:
Changed so that it's no longer buffered but has each msg send as a new goroutine